python/datasources/covid_tracking_project.py

```diff
@@ -14,5 +17,112 @@ def get_id():
     def get_table_name():
         return 'covid_tracking_project'
 
+    @staticmethod
+    def get_standard_columns():
+        """Returns a dict containing conversions from Covid Tracking Project's
+           race categories to their standardized values."""
+        return {
+            'aian': col_std.Race.AIAN.value,
+            'asian': col_std.Race.ASIAN.value,
+            'black': col_std.Race.BLACK.value,
+            'nhpi': col_std.Race.NHPI.value,
+            'white': col_std.Race.WHITE.value,
+            'multiracial': col_std.Race.MULTI.value,
+            'other': col_std.Race.OTHER.value,
+            'unknown': col_std.Race.UNKNOWN.value,
+            'ethnicity_hispanic': col_std.Race.HISP.value,
+            'ethnicity_nonhispanic': col_std.Race.NH.value,
+            'ethnicity_unknown': col_std.Race.ETHNICITY_UNKNOWN.value,
+            'total': col_std.Race.TOTAL.value
+        }
+
     def write_to_bq(self, dataset, gcs_bucket, **attrs):
-        logging.info("Covid Tracking Project write_to_bq not implemented yet")
+        filename = self.get_attr(attrs, 'filename')
+        metadata_table_id = self.get_attr(attrs, 'metadata_table_id')
+        table_name = self.get_attr(attrs, 'table_name')
+
+        df = gcs_to_bq_util.load_csv_as_dataframe(gcs_bucket, filename)
+        self.clean_frame_column_names(df)
+
+        # Massage the data into the standard format.
+        df.drop(columns=['cases_latinx', 'deaths_latinx',
+                         'hosp_latinx', 'tests_latinx'], inplace=True)
+        df = df.melt(id_vars=['date', 'state'])
+        df[['variable_type', 'race_and_ethnicity']] = df.variable.str.split(
+            "_", 1, expand=True)
+        df.drop('variable', axis=1, inplace=True)
+        df.rename(columns={'state': 'state_postal_abbreviation'}, inplace=True)
+        df.replace({'race_and_ethnicity': self.get_standard_columns()},
+                   inplace=True)
+
+        # Get the metadata table
+        metadata = self._download_metadata(metadata_table_id)
+        if len(metadata.index) == 0:
+            raise RuntimeError(
+                'BigQuery call to {} returned 0 rows'.format(metadata_table_id))
+
+        # Merge the tables
+        merged = pd.merge(
+            df, metadata, how='left',
+            left_on=['state_postal_abbreviation', 'variable_type'],
+            right_on=['state_postal_abbreviation', 'variable_type'])
+        # Rename combined race categories
+        self._rename_race_category(merged, 'reports_api', col_std.Race.ASIAN,
+                                   col_std.Race.API)
+        self._rename_race_category(merged, 'reports_ind', col_std.Race.AIAN,
+                                   col_std.Race.INDIGENOUS)
+
+        merged.drop(columns=['reports_api', 'reports_ind'], inplace=True)
```

Comment:
Yeah, this is the big one that I wanted to get your input on. Technically any of the metadata columns can vary between cases/deaths/hospitalizations/tests, and several of them do (e.g. race_ethnicity_combined, reports_ethnicity, reports_race, uses_api...) This will result in a proliferation of columns if we use a pivoted table. My reasoning for using cases/deaths/etc as a breakdown column instead was that it makes it easy to look up the data you want without needing to do string parsing on column names. It also makes joining with the ACS data more straightforward.

I haven't worked on the frontend at all, but based on Krista's talk it seems like we would need to add a new Breakdown class? I'm not sure how much work that would be but in my head this format seems easier to work with. If it's not feasible, I think that creating multiple tables might be slightly preferable

Reply:
I agree with the part about not having separate columns because joining with ACS gets complicated and I don't want to do string-parsing on column names.

A 4th alternative I just thought of is using nulls and add additional rows where they differ:
```
reports_combined | cases | deaths
true             | 400   | null
false            | null  | 20
```
This would allow joining with ACS smoothly, and we just filter out nulls first. Thoughts on that?

@kkatzen what do you think? To clarify, the format Jennie is suggesting here is:
```
state | race_and_ethnicity | variable | number
NY    | white              | cases    | 50
NY    | white              | deaths   | 50
NY    | white              | hosp     | 50
```
This would require some massaging on the frontend. I don't think it'd be terribly complicated to do that in the variable providers. But I'm hesitant to add more custom logic on the frontend than already exists.

My preference is this 4th alternative using additional rows, and my second preference is separate files. What do y'all think?

Reply:
As discussed offline, we will go for option 2, splitting into multiple tables. I opened issue #308